src/test/java/core/basesyntax/service/impl/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.service.ReaderService;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReaderServiceImplTest {
+    private static ReaderService readerService;
+
+    @BeforeAll
+    public static void init() {
+        readerService = new ReaderServiceImpl();
+    }
+
+    @Test
+    void reader_fileNotExist_notOk() {
```

Comment:
Lot's of methods names in whole implementation do not match the naming convention from the common mistakes:
`<methodUnderTest>_<state>_<expectedBehavior>`
so especially this test should have name:
`readFromFile_fileNotExist_notOk()`
It is redundant put name of the service class 'reader' to the method name because the test class should test only one service class. But in service class we can have several methods, so mention the which method is tested is meaningful information.